c8/client.py

```diff
@@ -2749,6 +2749,12 @@ def remove_key_value_pairs(self, name):
 
     # client.create_api_key
 
+    def get_jwt(self, email, password, tenant=None, username=None):
+        payload = {"email": email, "password": password}
+        if tenant is not None and username is not None:
+            payload.update({"tenant": tenant, "username": username})
```

Comment:
It's not tenant "and"  username.. we can pass either tenant, or  username or both or none.. so I guess two separate if statement would be needed..
```suggestion
        if username is not None:
            payload["username"] = username
```

Reply:
i added in that way because I understood from API REFERENCE that the input has to be or email, or tenant and username.
`After obtaining the token, REST API calls may be invoked by passing the obtained token in the REST header. Either email or both tenant and username are required. Add the following to your header when you make REST calls: "Authorization: bearer TOKEN"`

Reply:
Agreed, but here you have made email a mandatory parameter. So you are actually not handling that case.
So to handle that case, you will have to make email also an optional parameter and add "email is None" check to the "if" you have written. then your approach would be fine!

Reply:
I would suggest that since now we have more checks and error throwing to do.. you can keep all this inside "_get_auth_token" function inside "payload is not None" condition... in client.py simply pass the data in the payload json even if it's none and handle(checks, conditions, error) everything in "_get_auth_token"... What do you think? or you want to discuss this or have any doubts do let me know?